src/common/com/intellij/plugins/haxe/lang/psi/impl/AbstractHaxePsiClass.java

```diff
@@ -72,15 +72,29 @@ public HaxeNamedComponent getTypeComponent() {
   @NotNull
   @Override
   public String getQualifiedName() {
-    final String name = getName();
+    String name = getName();
     if (getParent() == null) {
       return name == null ? "" : name;
     }
+
+    if(name == null && this instanceof HaxeAnonymousType) {
+      // restore name from parent
+      HaxeTypedefDeclaration typedefDecl = (HaxeTypedefDeclaration)(getParent().getParent());
+      if(typedefDecl != null) {
+        name = typedefDecl.getName();
```

Comment:
HaxeAnonymousTypeImpl has not getName(), what is right because it's just part of declaration. So instead of modifying Psi Impl I've just made it externally for QName computing.